modules/space/kepler.cpp

```diff
@@ -677,6 +774,79 @@ std::vector<Parameters> readSbdbFile(const std::filesystem::path& file) {
     return result;
 }
 
+std::vector<Parameters> readMpcFile(const std::filesystem::path& file) {
+    ghoul_assert(std::filesystem::is_regular_file(file), "File must exist");
+
+    std::ifstream f = std::ifstream(file);
+
+    // Automatically detecting the header in an MPC file is unfortuntely not trivially
+    // The data lines in the MPC file format must be at least 160 character in length and
+    // none of the header lines (with one exception) encountered thus far are less than
+    // these 160 characters long. The exception is a line exactly 160 characters long with
+    // all `-` characters as a delimiter between header and data
+    std::vector<Parameters> result;
+    std::string line;
+    int i = 0;
+    while (ghoul::getline(f, line)) {
+        i++;
+
+        if (line.size() < 160) {
+            // The line is too short to be a data line
+            continue;
+        }
+        if (line.starts_with("------------------")) {
+            // It is the special case of the header seperator
+            continue;
+        }
+
+        std::string designation = line.substr(0, 6);
+
+        // We skip over the definitions of the magnitude and slope since we are not using
+        // those values anyway
+        line = line.substr(20);
+
+        // If we get this far, we should be in the data segment of the file
+        auto initial = scn::scan<
+            std::string, double, double, double, double, double, double, double>
+            (
+                line, "{} {} {} {} {} {} {} {}"
+            );
+        if (!initial) {
+            throw ghoul::RuntimeError(std::format(
+                "Unable to parse initial block of line {} in data file '{}'. {}",
+                i, file, line
+            ));
+        }
+
+        auto& [epoch, meanAnomaly, argPeriapsis, ascNode, inclination, eccentricity,
+            meanMotion, semiMajorAxis] = initial->values();
+
+        std::string name = designation;
+        if (line.size() >= 194) {
+            name = line.substr(166, 28);
```

Comment:
Without knowing anything about the file formats, these are some wild magic numbers. Is it worth labeling them?

Reply:
Or add a comment about it